expr/expr.go

```diff
@@ -3266,6 +3267,82 @@ func EvalExpr(e *expr, from, until int32, values map[MetricRequest][]*MetricData
 		}
 
 		return results, nil
+
+	case "cactiStyle": // cactiStyle(seriesList, system=None, units=None)
+		// Get the series data
+		original, err := getSeriesArg(e.args[0], from, until, values)
+		if err != nil {
+			return nil, err
+		}
+
+		// Get the arguments
+		system, err := getStringNamedOrPosArgDefault(e, "system", 1, "")
+		if err != nil {
+			return nil, err
+		}
+		unit, err := getStringNamedOrPosArgDefault(e, "units", 2, "")
+		if err != nil {
+			return nil, err
+		}
+
+		// Deal with each of the series
+		for _, a := range original {
+			// Calculate min, max, current
+			//
+			// This saves calling summarizeValues 3 times and looping over
+			// the metrics 3 times
+			//
+			// For min:
+			// Ignoring any absent values and inf (if we have a value)
+			// Using summarizeValues("min", ...) results in incorrect values, when absent
+			// values are present
+			//
+			val_min := math.Inf(-1)
+			val_current := math.Inf(-1)
+			val_max := math.Inf(-1)
+			for i, av := range a.GetValues() {
+				if !a.IsAbsent[i] && (val_min == math.Inf(-1) || av < val_min) {
+					val_min = av
+				}
+				if !a.IsAbsent[i] && av > val_max {
+					val_max = av
+				}
+				val_current = av
+			}
```

Comment:
This looks correct. I have some style comments:

* Go projects generally prefer camelCase to snake_case (from golint). val_min -> valMin
* You check `!a.IsAbsent[i]` in both if blocks which trips my **D**on't **R**epeat **Y**ourself sense a little.
* Checking `val_min == math.Inf(-1)` could be more elegantly/readably achieved by initializing `val_min := math.Inf(1)` (+Inf). Then the == check should not be necessary.

There are also math.Min() and math.Max() functions which probably add a small amount of overhead (edge case checking, function call, etc) but aid readability.

With those comments here's what I envision:
```go
    vMin := math.Inf(1)
    vCurrent := math.NaN()
    vMax := math.Inf(-1)
    for i, av := range a.GetValues() {
        if a.IsAbsent[i] {
            continue
        }
        vMin = math.Min(vMin, av)
        vMax = math.Max(vMax, av)
        vCurrent = av
    }
    // error handling if math.IsNaN(vCurrent) meaning we never saw a valid value
```